pkg/metadata/inventories/host_metadata.go

```diff
@@ -120,14 +120,16 @@ func getHostMetadata() *HostMetadata {
 		metadata.CPUArchitecture = platformInfo.HardwarePlatform
 	}
 
-	memoryInfo, warnings, err := memoryGet()
+	memoryInfo := memoryGet()
+	_, warnings, err = memoryInfo.AsJSON()
 	if err != nil {
 		logErrorf("failed to retrieve host memory metadata from gohai: %s", err) //nolint:errcheck
 	} else {
 		logWarnings(warnings)
 
-		metadata.MemoryTotalKb = memoryInfo.TotalBytes / 1024
-		metadata.MemorySwapTotalKb = memoryInfo.SwapTotalBytes / 1024
+		memoryTotalKb, _ := memoryInfo.TotalBytes.Value()
+		metadata.MemoryTotalKb = memoryTotalKb / 1024
+		metadata.MemorySwapTotalKb, _ = memoryInfo.SwapTotalKb.Value()
 	}
```

Comment:
This is not really an idiomatic way to use the new API but I tried to minimize the changes to the code outside Gohai.